sharktank/sharktank/models/llama/llama.py

```diff
@@ -129,20 +129,25 @@ def prefill(
         # [bs, batch_seq_len]
         tokens: Union[torch.Tensor, ReplicatedTensor],
         *,
+        # [bs, batch_seq_len]
+        input_mask: Union[torch.Tensor, ReplicatedTensor],
         # [1, 1, batch_seq_len, batch_seq_len]
         attention_mask: Optional[Union[torch.Tensor, ReplicatedTensor]],
         # [bs, batch_seq_len // block_seq_stride]
         seq_block_ids: Union[torch.Tensor, ReplicatedTensor],
         cache_state: list[Union[torch.Tensor, SplitPrimitiveTensor]],
     ):
         self._assert_device(tokens)
+        self._assert_device(input_mask)
         self._assert_device(attention_mask, dtype=self.activation_dtype)
         self._assert_device(seq_block_ids)
         self._assert_device(*cache_state, dtype=self.activation_dtype)
 
         h = self.token_embedding(tokens)
         self.trace_tensor("llama.token_embedding", h)
 
+        h *= input_mask.unsqueeze(-1)
```

Comment:
I don't think that if we assume this should somehow substitute the attention mask the math checks out.